rust-src/concordium_base/src/aggregate_sig/ffi.rs

```diff
@@ -77,7 +77,7 @@ pub extern "C" fn bls_sign(
     m_len: size_t,
     sk_ptr: *mut SecretKey<Bls12>,
 ) -> *mut Signature<Bls12> {
-    let m_len = m_len as usize;
+    let m_len = m_len;
```

Comment:
With this change, the assignment seems entirely unnecessary, and similarly for the others in this file. However, I am not entirely sure that removing the cast is the right option. I think for all platforms we care about, we can rely on `usize` and `size_t` being the same, but it seems at least historically we have tried to avoid making this assumption. I think probably the thing to do is keep the cast and suppress the warning.

Reply:
https://docs.rs/libc/latest/libc/type.size_t.html

On all platforms that I can find there size_t is defined as usize (as in type alias). Since the only reason we need to convert is so we can index in Rust there does not seem to be a downside to avoiding this cast.

If at some point Rust and Haskell add support for a platform where size_t will not be usize then that will lead to a compilation error for this module. So I think it is safe to remove the cast.

Reply:
Good point, I removed them.